tests/testutils/src/main/kotlin/io/element/android/tests/testutils/WithFakeLifecycleOwner.kt

```diff
@@ -33,10 +38,15 @@ fun <T> withFakeLifecycleOwner(
 ): T {
     currentComposer.startProvider(LocalLifecycleOwner provides lifecycleOwner)
     val state = block()
-    currentComposer.endProviders()
+    currentComposer.endProvider()
     return state
 }
 
+/**
+ * Test a [Presenter] with a fake [LifecycleOwner].
+ *
+ * **WARNING: DO NOT USE OUTSIDE TESTS.**
+ */
```

Comment:
Since this module is a test module, no prod module should have a dependency on it. Being able to use this method from a production module will reveal that there is an issue in the module dependencies. But it's probably safer to keep this warning anyway.